### PR TITLE
Updates build system

### PR DESCRIPTION
**Pull request by hughrawlinson**
Some more changes to the build system are in this pull request
* I've switched from using gulp/babel to browserify/babel.
  I've successfully compiled for browser, I just need to go through the
  actual source and fix the bugs I introduced by porting to ES6.
* We use `make` to build the package.
  Make calls out to gulp to do the node build (which doesn't quite work
  yet), and browserify to do the browser build (as well as a small gulp
  task to minify/uglify the source and generate source maps.

I will integrate browserify into the gulp pipeline so that we'll end up
with a structure something like:

* Gulp Tasks:
  * Build for Node
    * Transpiles to ES6
    * Writes to `dist/node/*` (preserving directory structure)
  * Build for Web
    * Concatenates
    * Transpiles to ES5
    * Writes to `dist/browser/meyda.js` along with sourcemap
    * uglifies
    * Writes to `dist/browser/meyda.min.js` along with sourcemap
* Make Tasks
  * `clean`
    * obviously
  * `build`
    * builds all, prepares build directory structure
  * `buildDev`
    * Some watching functionality, may be tough

**TLDR** use `make build` to make the distribution files once you've
made changes to the source.

### CODE REVIEW

**Comment by hughrawlinson** [comment]
Fixes #73, #60, and #70, so far.
